modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/domain/DefaultDomainFactory.java

```diff
@@ -159,7 +141,34 @@ public Domain createArtifact(File domainLocation, Optional<Properties> deploymen
       throw new IllegalArgumentException("Mule domain name may not contain spaces: " + domainName);
     }
 
-    return createArtitact(findDomain(domainName, deploymentProperties));
+    DomainDescriptor domainDescriptor = findDomain(domainName, deploymentProperties);
 
+    List<ArtifactPluginDescriptor> artifactPluginDescriptors =
+        domainDescriptor.getPlugins().stream().collect(Collectors.toList());
+    List<ArtifactPluginDescriptor> resolvedArtifactPluginDescriptors =
+        pluginDependenciesResolver.resolve(artifactPluginDescriptors);
+
+    DomainClassLoaderBuilder artifactClassLoaderBuilder =
+        domainClassLoaderBuilderFactory.createArtifactClassLoaderBuilder();
+    MuleDeployableArtifactClassLoader domainClassLoader =
+        artifactClassLoaderBuilder
+            .addArtifactPluginDescriptors(resolvedArtifactPluginDescriptors.toArray(new ArtifactPluginDescriptor[0]))
```

Comment:
`new ArtifactPluginDescriptor[0]` ?  I would expect a `[1]` instead. Anyway, I believe that the best practice is to use `ArtiactPluginDescriptor[]::new` and the stream automatically dims the array properly

Reply:
It doesn't work. Changing it to new ArtifactPluginDescriptor[resolvedArtifactPluginDescriptors.size()]